types/detect/detect.go

```diff
@@ -67,3 +70,27 @@ type Logger interface {
 	Warnw(format string, v ...interface{})
 	Errorw(format string, v ...interface{})
 }
+
+type DataSource interface {
+	// Get a value from the data source. Make sure the key matches one of the keys allowed in Keys.
+	// The following errors should be returned for the appropriate cases:
+	//
+	//	- ErrDataNotFound - When the key does not match to any existing data
+	//	- ErrKeyNotSupported - When the key used does not match to a support key
+	//	- Otherwise errors may vary.
+	Get(interface{}) (map[string]interface{}, error)
+	// Version of the data fetched. Whenever the schema has a breaking change the version should be incremented.
+	// Consumers of the data source should verify they are running against a support version before using it.
+	Version() uint
+	// The types of keys the data source supports.
+	Keys() []string
+	// JSON Schema of the data source's result. All Get results should conform to the schema described.
+	Schema() string
```

Comment:
Last question I guess. When are you planning on checking for conformity ? (to check if schema has changed, why wouldn't the version be enough and "a contract" between user and the data source) ? Not saying it is a bad idea to have the Schema() method, just trying to understand your thoughts!

Reply:
I'm not, this is more of a documentation in code thing that data source creators should add. I'm toying with adding a "DataSourceManager" which would proxy calls through it to data sources (this may be needed for "tracee analyze" compatibility later so we can export data source history), and then we could validate there, but not sure if it's a good idea performance-wise.